CMakeLists.txt

```diff
@@ -449,6 +449,7 @@ IF(HAVE_INTRIN_H)
         }" HAVE_BITSCANFORWARD_INTRINSIC)
 ENDIF()
 
+CHECK_SYMBOL_EXISTS(aligned_alloc   stdlib.h HAVE_ALIGNED_ALLOC)
```

Comment:
This unfortunately only tests under C, and some systems may expose `aligned_alloc` for C11, but not for C++ prior to C++17. And even if it is available in C++, it's technically `std::aligned_alloc`, just `aligned_alloc` isn't guaranteed to work (in practice it probably will, but I don't think it has to).

Reply:
Yes, it's right. However, almalloc.cpp uses `aligned_alloc` in C, not `std::aligned_alloc` in CPP.
So I think that if we make it more strict, we should check c++17 flag and check the existence of `std::aligned_alloc` function, then use `std::aligned_alloc` instead, or checking then using `aligned_alloc` in `stdlib.h` may be enough.